renku/core/commands/dataset.py

```diff
@@ -182,6 +190,24 @@ def add_to_dataset(
             'Cannot add multiple URLs with --source or --destination'
         )
 
+    if interactive:
+        if total_size is None:
+            total_size = 0
+            for url in urls:
+                try:
+                    with requests.get(url, stream=True) as r:
+                        total_size += int(r.headers.get('content-length', 0))
+                except requests.exceptions.RequestException:
+                    pass
+        usage = shutil.disk_usage(client.path)
+
+        if total_size > usage.free:
+            GB = 2**30
+            message = 'There isn\'t enough disk space ' \
+                      '(required: {:.2f} GB/available: {:.2f} GB). ' \
+                      'Continue anyway?'.format(total_size/GB, usage.free/GB)
```

Comment:
I'm just wondering. Is there a use case for continue in such cases when there is not enough disk space? Doesn't it make more sense just to abort any further computation in case on insufficient disk space?

Reply:
I agree, abort or make it a loop that rechecks if the user freed up some space in the meantime

Reply:
👍 The operation is aborted if there is not enough disk space.